sdk/core/azure-core-amqp/test/ut/link_tests.cpp

```diff
@@ -176,43 +185,76 @@ namespace Azure { namespace Core { namespace Amqp { namespace Tests {
         Azure::Core::Amqp::_internal::Session const& session,
         Azure::Core::Amqp::_internal::LinkEndpoint& newLinkInstance,
         std::string const& name,
-        Azure::Core::Amqp::_internal::SessionRole,
+        Azure::Core::Amqp::_internal::SessionRole sessionRole,
         Azure::Core::Amqp::Models::AmqpValue const& source,
         Azure::Core::Amqp::Models::AmqpValue const& target,
         Azure::Core::Amqp::Models::AmqpValue const&) override
     {
       GTEST_LOG_(INFO) << "OnLinkAttached - Link attached to session.";
-      auto newLink = std::make_unique<Azure::Core::Amqp::_detail::Link>(
-          session,
-          newLinkInstance,
-          name,
-          Azure::Core::Amqp::_internal::SessionRole::Receiver,
-          static_cast<std::string>(source),
-          static_cast<std::string>(target));
-      m_receiveLinkQueue.CompleteOperation(std::move(newLink));
+      std::unique_ptr<Azure::Core::Amqp::_detail::Link> newLink;
+      if (sessionRole == SessionRole::Sender)
+      {
+        newLink = std::make_unique<Azure::Core::Amqp::_detail::Link>(
+            session,
+            newLinkInstance,
+            name,
+            Azure::Core::Amqp::_internal::SessionRole::Receiver,
+            source,
+            target);
+      }
+      else
+      {
+        newLink = std::make_unique<Azure::Core::Amqp::_detail::Link>(
+            session,
+            newLinkInstance,
+            name,
+            Azure::Core::Amqp::_internal::SessionRole::Sender,
+            source,
+            target);
+      }
+      m_link = std::move(newLink);
+      m_receiveLinkQueue.CompleteOperation(true);
 
       return true;
     }
 
   public:
     LinkSocketListenerEvents() {}
-    std::shared_ptr<Connection> WaitForConnection(
+    bool WaitForConnection(
         Azure::Core::Amqp::Network::_detail::SocketListener const& listener,
         Azure::Core::Context const& context)
     {
       auto result = m_listeningQueue.WaitForPolledResult(context, listener);
-      return std::move(std::get<0>(*result));
+      return std::get<0>(*result);
     }
-    std::unique_ptr<Session> WaitForSession(Azure::Core::Context const& context)
+    bool WaitForSession(Azure::Core::Context const& context)
     {
       auto result = m_listeningSessionQueue.WaitForResult(context);
-      return std::move(std::get<0>(*result));
+      return std::get<0>(*result);
     }
-    std::unique_ptr<Azure::Core::Amqp::_detail::Link> WaitForLink(
-        Azure::Core::Context const& context)
+    bool WaitForLink(Azure::Core::Context const& context)
     {
       auto result = m_receiveLinkQueue.WaitForResult(context);
-      return std::move(std::get<0>(*result));
+      return std::get<0>(*result);
+    }
+
+    void Cleanup()
+    {
+      if (m_link)
+      {
+        //        m_link->Detach(true, {}, {}, {});
```

Comment:
```suggestion
        // m_link->Detach(true, {}, {}, {});
```
Or, if it is not needed, better remove it?